src/popup/pages/Names/AuctionList.vue

```diff
@@ -40,58 +40,83 @@
   </div>
 </template>
 
-<script>
-import { mapGetters } from 'vuex';
-import { pick } from 'lodash-es';
-import { watchUntilTruthy, blocksToRelativeTime } from '../../utils';
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  onMounted,
+  ref,
+} from '@vue/composition-api';
+import { watchUntilTruthy, blocksToRelativeTime, rxJsObservableToVueState } from '../../utils';
+import { useGetter } from '../../../composables';
+import { IActiveAuction } from '../../../types';
+
 import Filters from '../../components/Filters.vue';
 import NameRow from '../../components/NameRow.vue';
 import TokenAmount from '../../components/TokenAmount.vue';
-import AnimatedSpinner from '../../../icons/animated-spinner.svg?skip-optimize';
 import RegisterName from '../../components/RegisterName.vue';
+import AnimatedSpinner from '../../../icons/animated-spinner.svg?skip-optimize';
 
-export default {
+export default defineComponent({
   components: {
     Filters,
     NameRow,
     TokenAmount,
     AnimatedSpinner,
     RegisterName,
   },
-  data: () => ({
-    displayMode: { sort: 'soonest', rotated: false },
-    activeAuctions: [],
-    filters: { soonest: { rotated: false }, bid: { rotated: false }, length: { rotated: false } },
-    loading: false,
-  }),
-  subscriptions() {
-    return pick(this.$store.state.observables, ['topBlockHeight']);
-  },
-  computed: {
-    ...mapGetters(['getNameFee']),
-    auctions() {
-      return [...this.activeAuctions].sort((a, b) => {
-        switch (this.displayMode.sort) {
-          case 'length':
-            return a.name.length - b.name.length;
-          case 'bid':
-            return a.lastBid.nameFee - b.lastBid.nameFee;
-          default:
-            return 1;
-        }
-      }).sort(() => (this.displayMode.rotated ? -1 : 1));
-    },
-  },
-  async mounted() {
-    this.loading = true;
-    await watchUntilTruthy(() => this.$store.state.middleware);
-    this.activeAuctions = await this.$store.dispatch('names/fetchAuctions');
-    this.loading = false;
+  setup(props, { root }) {
+    const loading = ref(false);
+    const displayMode = ref({ sort: 'soonest', rotated: false });
+    const activeAuctions = ref<IActiveAuction[]>([]);
+    const filters = ref({
+      soonest: { rotated: false },
+      bid: { rotated: false },
+      length: { rotated: false },
+    });
+
+    const getNameFee = useGetter('getNameFee');
+
+    const auctions = computed(
+      () => [...activeAuctions.value]
+        .sort((a, b) => {
+          switch (displayMode.value.sort) {
+            case 'length':
+              return a.name.length - b.name.length;
+            case 'bid':
+              return parseInt(a.lastBid.nameFee, 10) - parseInt(b.lastBid.nameFee, 10);
+            default:
+              return 1;
+          }
+        })
+        .sort(() => (displayMode.value.rotated ? -1 : 1)),
+    );
```

Comment:
What do you think of using constants for the filter methods?

```
const SORT_MODE = {
     soonest: 'soonest',
     bid: 'bid',
     length: 'length'
   }
```

```suggestion 
    const displayMode = ref({ sort: SORT_MODE.soonest, rotated: false });
    const activeAuctions = ref<IActiveAuction[]>([]);
    const filters = ref({
      [SORT_MODE.soonest]: { rotated: false },
      [SORT_MODE.bid]: { rotated: false },
      [SORT_MODE.length]: { rotated: false },
    });

    const getNameFee = useGetter('getNameFee');

    const auctions = computed(
      () => [...activeAuctions.value]
        .sort((a, b) => {
          switch (displayMode.value.sort) {
            case [SORT_MODE.length]:
              return a.name.length - b.name.length;
            case [SORT_MODE.bid]:
              return parseInt(a.lastBid.nameFee, 10) - parseInt(b.lastBid.nameFee, 10);
            default:
              return 1;
          }
        })
        .sort(() => (displayMode.value.rotated ? -1 : 1)),
    );
```

Reply:
Makes sense. I spend some time and it turns out that the best for us would be to convert the Filters component to TypeScript also. Please review the recent changes.